app/javascript/components/patient/FollowUpFlag.js

```diff
@@ -0,0 +1,283 @@
+        this.setState(state_updates);
+      }
+    } else {
+      if (this.props.patient.follow_up_reason) {
+        // If the monitoree has a follow up flag set, enable the clear flag option
+        state_updates.clear_flag_disabled = false;
+        state_updates.follow_up_reason = this.props.patient.follow_up_reason;
+        state_updates.follow_up_note = this.props.patient.follow_up_note;
+        this.setState(state_updates);
+      }
+    }
+  }
+
+  handleChange = event => {
+    let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+    if (event.target.id === 'follow_up_reason') {
+      this.setState({ follow_up_reason: value });
+    } else if (event.target.id === 'follow_up_note') {
+      this.setState({ follow_up_note: value });
+    } else if (event.target.id === 'apply_to_household') {
+      this.setState({ apply_to_household: value });
+    } else if (event.target.id === 'bulk_action_apply_to_household') {
+      this.setState({ bulk_action_apply_to_household: value });
+    } else if (event.target.id == 'set_flag_for_follow_up') {
+      this.setState({ clear_flag: false });
+    } else if (event.target.id == 'clear_flag_for_follow_up') {
+      this.setState({ clear_flag: true });
+    } else if (event.target.id == 'clear_flag_reason') {
+      this.setState({ clear_flag_reason: value });
+    }
+  };
+
+  handleApplyHouseholdChange = apply_to_household => {
+    const no_members_selected = apply_to_household && this.state.apply_to_household_ids.length === 0;
+    this.setState({ apply_to_household, no_members_selected });
+  };
+
+  handleApplyHouseholdIdsChange = apply_to_household_ids => {
+    const no_members_selected = this.state.apply_to_household && apply_to_household_ids.length === 0;
+    this.setState({ apply_to_household_ids, no_members_selected });
+  };
+
+  // Makes a POST to update the follow-up flag for the current patient.
+  submit = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+
+      // Different POST requests are used depending on whether this action is triggered via the dashboard or monitoree page
+      if (this.props.bulk_action) {
+        let idArray = this.props.patients.map(x => x['id']);
+        axios
+          .post(window.BASE_PATH + '/patients/bulk_edit', {
+            ids: idArray,
+            bulk_edit_type: 'follow-up',
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.bulk_action_apply_to_household,
+          })
+          .then(() => {
+            location.href = window.BASE_PATH;
+          })
+          .catch(error => {
+            reportError(error);
+            this.setState({ loading: false });
```

Comment:
do we want to renable the buttons here? or force the user to refresh?

Reply:
I just tested this out and it doesn't make sense to reenable to the button on error because they could just keep hitting it and get spamming errors.  Can we remove this and have a helpful error message in the toast?